src/main/ruby/truffleruby/core/integer.rb

```diff
@@ -38,6 +38,7 @@
 Object.deprecate_constant :Fixnum, :Bignum
 
 class Integer < Numeric
+  FIXNUM_MAX = 0x3fffffff
```

Comment:
@eregon not sure that it's safe to hardcode this value, but I did't find better solution, mb you know?

Reply:
It seems odd to me we even need this value to implement Integer#[], do you know which arguments return it?

Reply:
We don't need this anymore.